iaso/api/groups.py

```diff
@@ -59,16 +59,31 @@ class Meta:
     created_at = TimestampField(read_only=True)
     updated_at = TimestampField(read_only=True)
 
+    def validate(self, attrs):
+        default_version = self._fetch_user_default_source_version()
+        if "source_ref" in attrs:
+            # Check if the source_ref is already used by another group
+            existing_group = Group.objects.filter(
+                source_ref=attrs["source_ref"], source_version=default_version
+            ).first()
+            if existing_group:
```

Comment:
I think you can also do it in this way which is even faster than fetching the first instance:

```suggestion
            existing_group = Group.objects.filter(
                source_ref=attrs["source_ref"], source_version=default_version
            )
            if existing_group.exists():
```